File: 14 September 2017/py/sieve.py
def generate_sieve(n):
    lookup = (*range(2, n),)
    result = lookup[:]

    for x in lookup:
        result = (*filter(lambda y: x == y or y % x != 0, result),)

    return result

limit = 102
sieve = generate_sieve(limit)

print(sieve)

File: 14 September 2017/clj/sieve.clj
(defn generate-sieve
  ([n] (generate-sieve
         (range 2 n) (int (Math/sqrt n))))
  ([sieve x]
          (if (< x 2) sieve
              (generate-sieve
                (filter #(or (> (mod % x) 0)
                             (= % x))
                        sieve) (dec x)))))

(let [x 102]
     (generate-sieve x))

